src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +165,999 @@ static DamageTable D_80A63CE0 = {
+}
+
+void EnBal_FloatUpToSky(EnBal* this, PlayState* play) {
+    f32 cosFactor;
+    f32 sinFactor;
+
+    if (this->timer < 500) {
+        cosFactor = (Math_CosS(this->skyFloatPhase) * 0.1f) + 1.0f;
+        this->balloonScale.z = cosFactor;
+        this->balloonScale.y = cosFactor;
+        sinFactor = (Math_SinS(this->skyFloatPhase) * 0.1f) + 1.0f;
+        this->balloonScale.x = sinFactor * sinFactor;
+        this->picto.actor.world.pos.y *= 500 - this->timer;
+        this->picto.actor.world.pos.y +=
+            this->timer * (this->picto.actor.home.pos.y + (50.0f * Math_SinS(this->skyFloatPhase)));
+        this->picto.actor.world.pos.y *= 0.002f;
+
+        if (this->timer < 100) {
+            this->timer++;
+        } else {
+            this->timer += 10;
+        }
+        this->skyFloatPhase += 800;
+    } else {
+        EnBal_SetupFloatIdle(this);
+    }
+}
+
+void EnBal_SetupGroundIdle(EnBal* this) {
+    if (this->actionFunc == EnBal_FallFromSky) {
+        this->forceEyesShut = false;
+        this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+        this->idleAnimStage = EN_BAL_IDLESTAGE_WAIT;
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, EN_BAL_ANIM_WAIT);
+    } else if (Rand_Next() & 1) {
+        this->idleAnimStage = EN_BAL_IDLESTAGE_ACTIVITY;
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, EN_BAL_ANIM_TALK_IDLE);
+    } else {
+        this->idleAnimStage = EN_BAL_IDLESTAGE_ACTIVITY;
+        this->forceEyesShut = true;
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, EN_BAL_ANIM_SPIN);
+    }
+
+    this->watchTarget = EN_BAL_WATCHTRG_NONE;
+    if (this->location == EN_BAL_LOC_CLOCK_TOWN) {
+        if (!gSaveContext.save.saveInfo.playerData.isMagicAcquired) {
+            // Effectively turn off reinflation timer by setting above 300
+            this->timer = 301;
+        } else if (this->inflateEarly == true) {
+            // If this field is set, reduce time to balloon reinflation.
+            this->inflateEarly = false;
+            this->timer = 290;
+        } else {
+            this->timer = 0;
+        }
+    } else {
+        this->timer = 0;
+    }
+
+    this->actionFunc = EnBal_GroundIdle;
+}
+
+void EnBal_GroundIdle(EnBal* this, PlayState* play) {
+    Player* player;
+    u8 playerForm;
+
+    player = GET_PLAYER(play);
+    if (this->timer == 300) {
+        if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+            EnBal_SetupInflateBalloon(this);
+            return; // Figure out how to move this...
+        }
+    } else if (this->timer < 300) {
+        this->timer++;
+    }
+
+    if (Actor_ProcessTalkRequest(&this->picto.actor, &play->state)) {
+        this->forceEyesShut = false;
+        this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_10_01)) {
+            playerForm = gSaveContext.save.playerForm;
+            if (EnBal_GetRecognizedPlayerForm() == playerForm) {
```

Comment:
```suggestion
            if (EnBal_GetRecognizedPlayerForm() == GET_PLAYER_FORM) {
```
GET_PLAYER_FORM adds a (void) 0, trick that may remove the need for the temp. If not could you post a scratch?

Reply:
I tried making this change (and removing the playerForm declaration), but it didn't match.
Here's the original scratch I used for this function. I'm happy to gen one with the updated context/names if you would prefer that, but I gotta dig up the asm etc etc, so this is faster.
https://decomp.me/scratch/DSSXH

Reply:
https://decomp.me/scratch/Zg7Wl seems to work for me, you just have to flip the condition as well.
```c
GET_PLAYER_FORM == EnBal_GetRecognizedPlayerForm()
```